### PR TITLE
Pass arguments to CatalogClient by value

### PR DESCRIPTION
**Pull request by diachenko-mischa**
Pass arguments to methods of CatalogClient by value and move them inside.

Relates-to: OLPEDGE-798
Signed-off-by: Diachenko Mykahilo <ext-mykhailo.z.diachenko@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h [hunk 1]
```diff
@@ -144,9 +142,8 @@ class DATASERVICE_READ_API CatalogClient final {
    * @deprecated Will be removed in 1.0
    */
   OLP_SDK_DEPRECATED("Will be removed in 1.0")
-  client::CancellationToken GetPartitions(
-      const PartitionsRequest& request,
-      const PartitionsResponseCallback& callback);
+  client::CancellationToken GetPartitions(PartitionsRequest request,
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h]: This one should be removed before 1.0 so you dont need to update it. Same for all deprecated methods marked for 1.0

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h [hunk 2]
```diff
@@ -75,8 +75,8 @@ class DATASERVICE_READ_API CatalogClient final {
    * available, or an error is encountered.
    * @return A token that can be used to cancel this request
    */
-  client::CancellationToken GetCatalog(const CatalogRequest& request,
-                                       const CatalogResponseCallback& callback);
+  client::CancellationToken GetCatalog(CatalogRequest request,
```
[the remark this answers is not in the document]
Comment (diachenko-mischa) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h]: Request types are forward declared in CatalogClient.h, and in places, where CatalogClient is used, headers for corresponding request types are included.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h (hunk 1 above)
Comment (diachenko-mischa) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h]: Done

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (diachenko-mischa) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h]: Fix for includes https://github.com/heremaps/here-olp-sdk-cpp/pull/384

**Pull request merged**